internal/langserver/diagnostics/diagnostics.go

```diff
@@ -16,26 +16,30 @@ import (
 type diagContext struct {
 	ctx    context.Context
 	uri    lsp.DocumentURI
-	diags  hcl.Diagnostics
 	source string
+	diags  []lsp.Diagnostic
 }
 
 // Notifier is a type responsible for queueing hcl diagnostics to be converted
 // and sent to the client
 type Notifier struct {
+	logger         *log.Logger
 	sessCtx        context.Context
 	diags          chan diagContext
+	diagsCache     map[string]map[string][]lsp.Diagnostic
```

Comment:
Just for readability, would you mind creating some custom types here and compose the final `diagsCache` type from these?

For example:
```go
type fileURI string
type diagnosticSource string
type fileDiagnostics map[diagnosticSource][]lsp.Diagnostic

diagsCache map[fileURI]fileDiagnostics